dev/tools/localization/bin/gen_date_localizations.dart

```diff
@@ -46,32 +46,38 @@ Future<void> main(List<String> rawArgs) async {
 
   final bool writeToFile = parseArgs(rawArgs).writeToFile;
 
-  final File dotPackagesFile = File(path.join('packages', 'flutter_localizations', '.packages'));
-  final bool dotPackagesExists = dotPackagesFile.existsSync();
+  final File packageConfigFile = File(path.join('packages', 'flutter_localizations', '.dart_tool', 'package_config.json'));
+  final bool packageConfigExists = packageConfigFile.existsSync();
 
-  if (!dotPackagesExists) {
+  if (!packageConfigExists) {
     exitWithError(
-      'File not found: ${dotPackagesFile.path}. $_kCommandName must be run '
+      'File not found: ${packageConfigFile.path}. $_kCommandName must be run '
       'after a successful "flutter update-packages".'
     );
   }
 
-  final String pathToIntl = dotPackagesFile
-    .readAsStringSync()
-    .split('\n')
-    .firstWhere(
-      (String line) => line.startsWith('intl:'),
-      orElse: () {
-        exitWithError('intl dependency not found in ${dotPackagesFile.path}');
-        return ''; // unreachable
-      },
-    )
-    .split(':')
-    .last;
-
-  final Directory dateSymbolsDirectory = Directory(path.join(pathToIntl, 'src', 'data', 'dates', 'symbols'));
+  final List<dynamic> packages = (
+    json.decode(packageConfigFile.readAsStringSync()) as Map<String, Object?>
+  )['packages']! as List<dynamic>;
+
+  String? pathToIntl;
+  for (final dynamic package in packages) {
+    final Map<String, dynamic> packageAsMap = package as Map<String, dynamic>;
+    if (packageAsMap['name'] == 'intl') {
+      pathToIntl = (packageAsMap['rootUri'] as String).substring(7);
```

Comment:
What does the magic number 7 mean here?

Reply:
Ah so the uris in package_config.json are formatted with `file://` prepended to each url. I couldn't figure out a better way to remove it but let me know if you have better ideas.

Reply:
Should this use `Uri.parse(uriPath).toFilePath()` to be a little more robust then?

Reply:
Done!